src/app/pages/details/common/get-this-title-files/give-before-pdf/use-give-dialog.tsx

```diff
@@ -73,12 +72,21 @@ export function useOpenGiveDialog() {
     return {GiveDialog, openGiveDialog};
 }
 
-function lookupVariant(warning: string, variant?: VariantValue, id?: string) {
+// This was a little bit clever, so typing became a problem
+function lookupVariant(warning: string, variantParams: {
+    id?: string;
+    variant?: string;
+}): [
+    Variant: (p: any) => React.JSX.Element, // eslint-disable-line @typescript-eslint/no-explicit-any
+    p: object
+] {
+    const {id, variant} = variantParams;
```

Comment:
Do we want to use a rest object for the `variantParams` to prevent `id` and `variant` being sent to the components?

Reply:
All of the components use `id` and two of them use `variant`. There's likely a better way to do the repetition-elimination I was doing here, but it hasn't become clear to me yet.